src/basic/log.h

```diff
@@ -367,3 +368,38 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
+typedef struct LogRateLimit {
+        int error;
+        RateLimit ratelimit;
+} LogRateLimit;
```

Comment:
We should save the level of the log message as well, it's not always guaranteed to be the same.

Reply:
Should we also reset the rate limit if the level changes?

Reply:
See my other comment, I think we should